servers/dsub/jobs/controllers/jobs_controller.py

```diff
@@ -121,21 +121,27 @@ def query_jobs(body):
         query.page_token) or (None, None)
     query.page_size = min(query.page_size or _DEFAULT_PAGE_SIZE,
                           _MAX_PAGE_SIZE)
+
+    query.submission = query.submission.replace(tzinfo=tzlocal()).replace(
+        microsecond=0) if query.submission else None
+    query.start = query.start.replace(tzinfo=tzlocal()).replace(
+        microsecond=0) if query.start else None
+    query.end = query.end.replace(tzinfo=tzlocal()).replace(
+        microsecond=0) if query.end else None
+
     if query.page_size < 0:
         raise BadRequest("The pageSize query parameter must be non-negative.")
-    # TODO(bryancrampton): Fix this to support querying by submission and so
-    # that query.start actually filters by start-time.
-    if query.start:
-        query.start = query.start.replace(tzinfo=tzlocal()).replace(
-            microsecond=0)
-        if create_time_max and query.start > create_time_max:
-            raise BadRequest(
-                "Invalid query: start date is invalid with pagination token.")
-    if query.end:
-        query.end = query.end.replace(tzinfo=tzlocal()).replace(microsecond=0)
-        if query.start and query.start >= query.end:
-            raise BadRequest(
-                "Invalid query: start date must precede end date.")
+    if query.submission and query.start and query.submission >= query.start:
+        raise BadRequest(
+            "Invalid query: submission date must precede start date.")
```

Comment:
I'd probably go with <= instead. "submitted and started after time t" seems like it should be a valid query, and there's nothing semantically incorrect about submit->start taking <1s.

Reply:
Good point, done.